src/connection.c

```diff
@@ -929,7 +929,7 @@ CONNECTION_HANDLE connection_create2(XIO_HANDLE xio, const char* hostname, const
     }
     else
     {
-        result = (CONNECTION_INSTANCE*)amqpalloc_malloc(sizeof(CONNECTION_INSTANCE));
+        result = (CONNECTION_INSTANCE*)amqpalloc_calloc(1, sizeof(CONNECTION_INSTANCE));
```

Comment:
Interesting.

Actually having a second look at this it seems that we should get the tick counter stamp when creating the connection rather than initialize with 0. Let me have another look at the code.